src/features/views/components/ViewColumnDialog/SurveyResponseConfig.tsx

```diff
@@ -0,0 +1,226 @@
+    selectedOption: string | SURVEY_QUESTION_OPTIONS
+  ) => {
+    if (selectedQuestion !== undefined && selectedQuestion !== null) {
+      if (selectedOption === SURVEY_QUESTION_OPTIONS.ALL_OPTIONS) {
+        return [
+          {
+            config: {
+              question_id: selectedQuestion.id,
+            },
+
+            title: selectedQuestion.question.question,
+            type: COLUMN_TYPE.SURVEY_OPTIONS,
+          },
+        ];
+      } else if (
+        selectedOption === SURVEY_QUESTION_OPTIONS.ALL_OPTIONS_SEPARATED
+      ) {
+        return selectedQuestion.question.options?.map((option) => {
+          return {
+            config: {
+              option_id: option.id,
+              survey_id: selectedQuestion.id,
+            },
+            title: option.text,
+            type: COLUMN_TYPE.SURVEY_OPTION,
+          };
+        });
+      } else {
+        const optionTitle = selectedQuestion.question.options?.find(
+          (option) => option.id === (selectedOption as unknown as number)
+        );
+        if (optionTitle === undefined) {
+          return [];
+        }
+        return [
+          {
+            config: {
+              option_id: selectedOption as unknown as number,
+              survey_id: selectedQuestion.id,
+            },
+            title: optionTitle.text,
+            type: COLUMN_TYPE.SURVEY_OPTION,
+          },
+        ];
+      }
+    }
+  };
+
+  return (
+    <ZUIQuery queries={{ surveysQuery }}>
+      {({ queries: { surveysQuery: successSurveysQuery } }) => {
+        const selectedSurvey = successSurveysQuery.data.find(
+          (survey) => survey.id == surveyId
+        );
+        const questionFromSurvey = selectedSurvey?.elements.filter(
+          (elem) => elem.type == ELEMENT_TYPE.QUESTION
+        );
+
+        return (
+          <FormControl sx={{ width: 300 }}>
+            <TextField
+              fullWidth
+              label={intl.formatMessage({
+                id: 'misc.views.columnDialog.choices.surveyResponse.surveyField',
+              })}
+              margin="normal"
+              onChange={onSurveyChange}
+              select
+              value={surveyId || ''}
+              variant="standard"
+            >
+              {successSurveysQuery.data.map((survey) => (
+                <MenuItem key={survey.id} value={survey.id}>
+                  {survey.title}
+                </MenuItem>
+              ))}
+            </TextField>
+            {surveyId ? (
+              <Autocomplete
+                disabled={!surveyId}
+                fullWidth
+                getOptionLabel={(option) => option.question.question}
+                onChange={(evt, value) => {
+                  if (value !== null) {
+                    setSelectedQuestion(value);
+                    const columns = makeColumnsText(value);
+                    if (columns !== undefined) {
+                      onOutputConfigured(columns);
+                    }
```

Comment:
I find this logic to be difficult to follow. Unless I'm mistaken, what happens here is that `makeColumnsText()` is invoked for any question, even if it's not a text question.

For non-text questions, the function will return `undefined`, so on the next line we check for undefined before proceeding.

I think we should make sure that it's a text question here, before even calling `makeColumnsText()` and I think `makeColumnsText()` should only accept text questions (if there is a type for them, otherwise let's ignore this.).

Reply:
Looking into the types more closely, I see that there is no special type for text based questions, so maybe this can be ignored for now. Or we can refactor it together.